src/runners/junitRunner/JUnitRunnerResultAnalyzer.ts

```diff
@@ -7,22 +7,26 @@ import { dataCache, ITestItemData } from '../../controller/testItemDataCache';
 import { createTestItem, updateOrCreateTestItem } from '../../controller/utils';
 import { IJavaTestItem, IRunTestContext, TestKind, TestLevel } from '../../types';
 import { RunnerResultAnalyzer } from '../baseRunner/RunnerResultAnalyzer';
-import { findTestLocation, setTestState, TestResultState } from '../utils';
+import { CurrentItemState, findTestLocation, setTestState, TestResultState } from '../utils';
+
 
 export class JUnitRunnerResultAnalyzer extends RunnerResultAnalyzer {
 
     private testOutputMapping: Map<string, ITestInfo> = new Map();
     private triggeredTestsMapping: Map<string, TestItem> = new Map();
-    private currentTestState: TestResultState;
+    private projectName: string;
+    private incompleteTestSuite: ITestInfo[] = [];
+
+    // tests may be run concurrently, so each item's current state needs to be remembered
+    private currentStates: Map<TestItem, CurrentItemState> = new Map();
+
+    // failure info for a test is received consecutively:
     private currentItem: TestItem | undefined;
```

Comment:
nit: how about give it a more specific name, something like: `tracingItem`?

Reply:
just changed the name 👍🏼